server/broadcast.go

```diff
@@ -604,3 +630,26 @@ func updateSession(sess *BroadcastSession, res *ReceivedTranscodeResult) *Broadc
 
 	return newSess
 }
+
+func refreshSession(sess *BroadcastSession) (*BroadcastSession, error) {
+	uri, err := url.Parse(sess.OrchestratorInfo.Transcoder)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), refreshTimeout)
+	defer cancel()
+
+	oInfo, err := getOrchestratorInfoRPC(ctx, sess.Broadcaster, uri)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create dummy result
+	res := &ReceivedTranscodeResult{
+		LatencyScore: sess.LatencyScore,
+		Info:         oInfo,
+	}
+
+	return updateSession(sess, res), nil
```

Comment:
Let's add a test that ensures that the returned session actually uses the latest `OrchestratorInfo` returned by the `getOrchestratorInfoRPC`

Reply:
fixed in 20340ceabc68452fb4c8914a568e965e1c5f5d30